clustered/common/src/main/java/org/ehcache/clustered/common/store/ElementBuilder.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.common.store;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Builds {@link Element}s
+ */
+public interface ElementBuilder {
+
+  /**
+   * Gets an updated {@link Element} from existing element
+   * with new payLoad
+   *
+   * @param element the old element
+   * @param payLoad the new payLoad
+   * @return the Element with a new payLoad
+   */
+  Element getElement(Element element, ByteBuffer payLoad);
```

Comment:
I do not understand the role of this method. From the data structure we plan to use, most often we will end up wanting to replace a sub `Chain` of `Element`s with a `Chain` of a single `Element`. And so I do not see where this would be used.